infrastructure/balsa_queue/balsa_queue.cc

```diff
@@ -7,23 +7,56 @@
 
 namespace jet {
 
+namespace {
+  double nanoseconds_to_seconds(double nanoseconds) {
```

Comment:
Is it possible for this to live in `Duration` with the other conversion functions?

Reply:
Also could we convert to `std::chrono` so that we get these sorts of conversions for free?

Reply:
Consider std::chrono since precision will be lost in the floating-point division

Reply:
I considered using std::chrono, but I wanted to get this PR up quickly so that we could start using it right away. Will take another look to see how quickly I can make the switch.